src/components/AsyncStream.jsx

```diff
@@ -74,4 +76,11 @@ const AsyncStream = ({
   return <LoadableStream />;
 };
 
-export default AsyncStream;
+
+export default compose(
+  lifecycle({
+    shouldComponentUpdate(nextProps){
+      return nextProps == this.Props;
```

Comment:
Does `this.Props` actually exist here? I added a `console.log()` here and it's just always `undefined`, so this expression will never be `true`.

Reply:
A consequence of this is that the channel/stream form in the header no longer works if you try to use it whilst already on a stream. This is because the router will pass new props to `AsyncStream` but this expression tells React that no update is necessary so nothing happens.

Reply:
My bad, it was late and I didn't test it properly, It should have been this.props.

I changed it to check if the pathname is different and now it works
`return nextProps.location.pathname != this.props.location.pathname;`